utils/optiwebp.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+
+##
+# This script uses cwebp to convert existing png/jpg images to webp, or reconvert existing webp images, to see if doing so meaningfully reduces their size.
+##
+
+import argparse
+import os
+from pathlib import Path
+import shutil
+import signal
+import subprocess
+import sys
+import time
+
+def signal_handler(signal, func):
+    print("\nProcessing interrupted.")
+    shutil.rmtree(options.tempdir)
+    sys.exit(1)
+
+signal.signal(signal.SIGINT, signal_handler)
+
+args = argparse.ArgumentParser()
+
+args.add_argument("--minpercent", default=90, help="The new file size must be reduced by to least this percentage compared to the original. Default 90%% (10%% reduction).")
+args.add_argument("--nodryrun", action="store_true", help="Whether to replace the original file with the converted one if its size is reduced enough. Does not replace by default.")
+args.add_argument("--quality", default="90", help="The quality to use for lossy compression. Defaults to 90.")
+args.add_argument("--repo", default=os.getcwd(), help="The root of the Wesnoth repository. Defaults to the current directory.")
+args.add_argument("--tempdir", default=os.path.join(os.getcwd(), "temp_webp"), help="The root of the Wesnoth repository. Defaults to 'temp_webp/' in the current directory.")
+
+options = args.parse_args()
+
+# test we're in a repo root
+if not os.path.isfile(os.path.join(options.repo, "changelog.md")):
+    print("Didn't find changelog.md so assuming this isn't the repo root, exiting.")
+    sys.exit(1)
+
+# test cwebp presence
+if subprocess.run(["cwebp", "-version"], stdout=subprocess.DEVNULL).returncode != 0:
+    print("cwebp executable not found in PATH, exiting.")
+    sys.exit(1)
+
+image_dirs = [
```

Comment:
Here you should use `os.path.join()` instead of slashes.

Reply:
Done.